test/clj/rems/test/api/applications.clj

```diff
@@ -839,28 +907,33 @@
         (is (= {:success true} (send-dynamic-command handler-id {:type :rems.workflow.dynamic/approve
                                                                  :application-id application-id
                                                                  :comment ""})))
-        (let [data (get-application handler-id application-id)]
-          (is (= {:id application-id
-                  :state "rems.workflow.dynamic/approved"}
-                 (select-keys (:application data) [:id :state])))
-          (is (= ["application.event/created"
-                  "application.event/draft-saved"
-                  "application.event/submitted"
-                  "application.event/decision-requested"
-                  "application.event/decided"
-                  "application.event/approved"]
-                 (map :event/type (get-in data [:application :dynamic-events]))))))
-      (testing "applicant cannot see all events"
-        (let [data (get-application user-id application-id)]
-          (is (= {:id application-id
-                  :state "rems.workflow.dynamic/approved"}
-                 (select-keys (:application data) [:id :state])))
-          (is (= ["application.event/created"
-                  "application.event/draft-saved"
-                  "application.event/submitted"
-                  ; decision-requested and decided should be missing here
-                  "application.event/approved"]
-                 (map :event/type (get-in data [:application :dynamic-events])))))))))
+        (let [handler-data (get-application handler-id application-id)
+              handler-event-types (map :event/type (get-in handler-data [:application :dynamic-events]))
+              applicant-data (get-application user-id application-id)
+              applicant-event-types (map :event/type (get-in applicant-data [:application :dynamic-events]))]
+          (testing "handler can see all events"
+            (is (= {:id application-id
+                    :state "rems.workflow.dynamic/approved"}
+                   (select-keys (:application handler-data) [:id :state])))
+            (is (= ["application.event/created"
+                    "application.event/draft-saved"
+                    "application.event/submitted"
+                    "application.event/comment-requested"
+                    "application.event/comment-requested"
+                    "application.event/commented"
+                    "application.event/commented"
+                    "application.event/decision-requested"
+                    "application.event/decided"
+                    "application.event/approved"]
+                   handler-event-types)))
+          (testing "applicant cannot see all events"
+            (is (= (filter ;; all decision and comment related events should be missing here
```

Comment:
I'd prefer writing out the list of expected events explicitly. Now there is a possibility of bugs in in the filter call. It's a question of "who tests the tests". The code has no obvious mistakes, but I much prefer it to obviously have no mistakes.

Reply:
I wanted to avoid the comment and have an explicit list of events both ways. We have had regressions in this functionality before (that's why check with a blacklist as well rather than just comment about it) and since it's about who can see what it should be coded defensively (i.e. whitelist).